tools/scons/site_scons/site_tools/emscripten/emscripten.py

```diff
@@ -31,21 +39,25 @@ def generate(env, emscripten_path=None, **kw):
               'EMCC_FORCE_STDLIBS', 'EMCC_ONLY_FORCED_STDLIBS', 'EM_PORTS', 'IDL_CHECKS', 'IDL_VERBOSE']:
     if os.environ.get(var):
       env['ENV'][var] = os.environ.get(var)
+
   try:
     emscPath = emscripten_path.abspath
   except:
     emscPath = emscripten_path
 
-  env.Replace(CC=os.path.join(emscPath, "emcc"))
-  env.Replace(CXX=os.path.join(emscPath, "em++"))
+  building_env_variables = get_building_env_variables(emscPath)
+  env['ENV'].update(building_env_variables)
+
+  env.Replace(CC=building_env_variables['CC'])
+  env.Replace(CXX=building_env_variables['CXX'])
   # LINK uses smark_link by default which will choose
   # either emcc or em++ depending on if there are any C++ sources
   # in the program, so no need to change that.
   # SHLINK and LDMODULE should use LINK so no
   # need to change them here
 
-  env.Replace(AR=os.path.join(emscPath, "emar"))
-  env.Replace(RANLIB=os.path.join(emscPath, "emranlib"))
+  env.Replace(AR=building_env_variables['AR'])
+  env.Replace(RANLIB=building_env_variables['RANLIB'])
 
   env.Replace(OBJSUFFIX=[".js", ".bc", ".o"][2])
   env.Replace(LIBSUFFIX=[".js", ".bc", ".o"][2])
```

Comment:
Just realised this is setting the file extension for static libraries from the default of `.a` to `.o`
That doesn't make any sense, does it?

Reply:
Static libraries should be `.o` yes.  This code looks very odd to me.  Why use those temporary arrays at all?

Reply:
(sorry they should be `.a`.  edited)